src/twisted/conch/avatar.py

```diff
@@ -6,12 +6,15 @@
 from twisted.conch.error import ConchError
 from twisted.conch.interfaces import IConchUser
 from twisted.conch.ssh.connection import OPEN_UNKNOWN_CHANNEL_TYPE
-from twisted.python import log
 from twisted.python.compat import nativeString
+from twisted.logger import Logger
+
 
 
 @implementer(IConchUser)
 class ConchUser:
+    log = Logger()
```

Comment:
Don't have time for a full review here but since this came across my field of vision:

Can you avoid exposing this as a public attribute by making it `_log`?